javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/injection/InjectionMode.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.extension.instrumentation.injection;
+
+public enum InjectionMode {
+  CLASS_ONLY
+  // TODO: implement the modes RESOURCE_ONLY and CLASS_AND_RESOURCE
```

Comment:
For proxy classes I think we could expose the original class bytes and we already have the code to do that. Implementing a custom `URLStreamHandler` is also fine if we want to expose the actual proxy bytes.

Reply:
I'd prefer to provide the actual bytecode for the proxies. Applications parsing the bytecode might otherwise trip over references to non-existing types from the method bodies. In addition there might be some non-public methods and fields present in the bytecode but not in the proxy-class, which could also cause hard to debug issues.

I've already tried the approach with a custom `URLStreamHandler` on a separate branch, it is not much code and should be easy to maintain.